docs/en/api/serial.rst

```diff
@@ -0,0 +1,666 @@
+  * ``UART_MODE_RS485_APP_CTRL`` (0x04) - Application-controlled RS485 mode (for testing)
+
+**Returns:** ``true`` if mode is set successfully, ``false`` otherwise.
+
+**Note:** For RS485 half-duplex mode, the RTS pin must be configured using ``setPins()`` to control the transceiver.
+
+setClockSource
+**************
+
+Sets the UART clock source. Must be called **before** ``begin()`` to take effect.
+
+.. code-block:: arduino
+
+    bool setClockSource(SerialClkSrc clkSrc);
+
+* ``clkSrc`` - Clock source:
+  
+  * ``UART_CLK_SRC_DEFAULT`` - Default clock source (varies by SoC)
+  * ``UART_CLK_SRC_APB`` - APB clock (ESP32, ESP32-S2, ESP32-C3, ESP32-S3)
+  * ``UART_CLK_SRC_PLL`` - PLL clock (ESP32-C2, ESP32-C5, ESP32-C6, ESP32-C61, ESP32-H2, ESP32-P4)
+  * ``UART_CLK_SRC_XTAL`` - XTAL clock (ESP32-C2, ESP32-C3, ESP32-C5, ESP32-C6, ESP32-C61, ESP32-H2, ESP32-S3, ESP32-P4)
+  * ``UART_CLK_SRC_RTC`` - RTC clock (ESP32-C2, ESP32-C3, ESP32-C5, ESP32-C6, ESP32-C61, ESP32-H2, ESP32-S3, ESP32-P4)
+  * ``UART_CLK_SRC_REF_TICK`` - REF_TICK clock (ESP32, ESP32-S2)
+
+**Note:**
+* Clock source availability varies by SoC
+* PLL frequency varies by SoC: ESP32-C2 (40 MHz), ESP32-H2 (48 MHz), ESP32-C5/C6/C61/P4 (80 MHz)
+* ESP32-C5, ESP32-C6, ESP32-C61, and ESP32-P4 have LP UART that uses only RTC_FAST or XTAL/2 as clock source
+* For ESP32 and ESP32-S2, REF_TICK is used by default for baud rates ≤ 250000 to avoid baud rate changes when CPU frequency changes, but this limits RX timeout to 1 symbol
+
+**Returns:** ``true`` if clock source is set successfully, ``false`` otherwise.
+
+setRxInvert
+***********
+
+Enables or disables RX signal inversion.
+
+.. code-block:: arduino
+
+    bool setRxInvert(bool invert);
+
+* ``invert`` - If ``true``, inverts the RX signal polarity
+
+**Returns:** ``true`` if inversion is set successfully, ``false`` otherwise.
+
+setTxInvert
+***********
+
+Enables or disables TX signal inversion.
+
+.. code-block:: arduino
+
+    bool setTxInvert(bool invert);
+
+* ``invert`` - If ``true``, inverts the TX signal polarity
+
+**Returns:** ``true`` if inversion is set successfully, ``false`` otherwise.
+
+setCtsInvert
+************
+
+Enables or disables CTS signal inversion.
+
+.. code-block:: arduino
+
+    bool setCtsInvert(bool invert);
+
+* ``invert`` - If ``true``, inverts the CTS signal polarity
+
+**Returns:** ``true`` if inversion is set successfully, ``false`` otherwise.
+
+setRtsInvert
+************
+
+Enables or disables RTS signal inversion.
+
+.. code-block:: arduino
+
+    bool setRtsInvert(bool invert);
+
+* ``invert`` - If ``true``, inverts the RTS signal polarity
+
+**Returns:** ``true`` if inversion is set successfully, ``false`` otherwise.
+
+setDebugOutput
+**************
+
+Enables or disables debug output on this Serial port.
+
+.. code-block:: arduino
+
+    void setDebugOutput(bool enable);
+
+* ``enable`` - If ``true``, enables debug output (ESP-IDF log messages will be sent to this Serial port)
```

Comment:
You see? Make all product names consistent.

Reply:
done